### ISSUE TITLE
Non-matching authentication token or forgotten authenticated app

### ISSUE THREAD

**Issue by christianlupus** [nextcloud/Android-SingleSignOn#254]
I have the problem, that there seems to be a problem with the SSO tokens on my girlfriend's android phone. She tried to use the notes app by @stefan-niedermann.

I post this issue here as I assume it is related to the SSO library.

When selecting an account from the list during initial logon she chooses the account from the main files app. After that the following debug information are generated:

```
App Version: 2.16.7
App Version Code: 2016007
App Flavor: fdroid

Files App Version Code: 30130190

---

OS Version: 3.18.35+(1488252935)
OS API Level: 24
Device: K6000_Plus
Manufacturer: OUKITEL
Model (and Product): K6000 Plus (K6000_Plus)

---

com.nextcloud.android.sso.exceptions.TokenMismatchException: Der Authentifizierungs-Token stimmt nicht überein oder dem Paketnamen fehlt die Berechtigung für diese Aktion
	at com.nextcloud.android.sso.api.AidlNetworkRequest.performNetworkRequestV2(AidlNetworkRequest.java:172)
	at com.nextcloud.android.sso.api.NextcloudAPI.performNetworkRequestV2(NextcloudAPI.java:148)
	at it.niedermann.owncloud.notes.persistence.SSOClient.requestFilesApp(SSOClient.java:27)
	at it.niedermann.owncloud.notes.persistence.CapabilitiesClient.getCapabilities(CapabilitiesClient.java:64)
	at it.niedermann.owncloud.notes.main.MainActivity.lambda$null$20$MainActivity(MainActivity.java:791)
	at it.niedermann.owncloud.notes.main.-$$Lambda$MainActivity$4HeEjJCK6xYZ8GdVmxdXI3hCjBw.run(lambda)
	at java.lang.Thread.run(Thread.java:761)
```

The translation of the error message from German would be something like *The authentication token does not match or the package name is missing the permission for this action*.

Additional information:
This could be related: She used to use the regular files app from NC to open and modify the `*.md` files on the phone manually. Here exists an issue as well: Sporadically (unfortunately) it is not possible to open the file. There is a symbol indicating to wait but no action seems to be done as this can take minutes without result (opening the file). The said file is only a few bytes, so it should not be a networking delay caused by vast amount of data.

**Comment by christianlupus** [comment]
I had to edit the original issue, as I misinterpreted the description from my GF.

**Comment by tobiasKaminsky** [comment]
@David-Development what happens if 3rd party app is installed, used, uninstalled, installed again?
Do we remove the old token somehow?

**Comment by David-Development** [comment]
@tobiasKaminsky the token is only temporarily stored in third party apps - so after the 3-party app is reinstalled, the token is gone inside the third party app (at least in the news app - I think this is also true for stefans apps). 

Old tokens are not cleared in the nextcloud files app. So the last valid token is stored in the nextcloud files app. Once the user authenticates the same account again, the previous token for the specific account will be deleted (see https://github.com/nextcloud/android/blob/d251fbb556c5e189c64d1842742f37772a5b2df3/src/main/java/com/owncloud/android/ui/activity/SsoGrantPermissionActivity.java#L180)

**Comment by christianlupus** [comment]
"Unfortunately" I can no longer confirm this issue. At least I was now able to log in successfully after creating a new account on the main NC app. In fact I was affected by #200. Therefore I needed to create a new account in the phone with the correct URL. 

I am not sure if this switching to a new account simply solved my problem her or just hid it. The old account is still in place (I configured only a second one) so I could switch that back for testing if need arises. Apart from that, I am not able to provide much feedback and I would be ok with closing this issue right now.
